src/modules/billing/controller.js

```diff
@@ -75,23 +78,23 @@ const getBatch = async request => {
 
 const getBatchInvoices = async request => {
   const { batchId } = request.params;
-  const invoices = await repos.billingInvoices.findByBatchId(batchId);
+  const invoices = await invoiceService.getInvoicesForBatch(batchId);
 
   return invoices.length
     ? envelope(invoices, true)
     : Boom.notFound(`No invoices found for batch with id: ${batchId}`);
 };
 
-const getInvoiceDetail = async request => {
-  const { invoiceId } = request.params;
-  const invoice = await repos.billingInvoices.getInvoiceDetail(invoiceId);
+const getBatchInvoiceDetail = async request => {
+  const { batchId, invoiceId } = request.params;
+  const invoice = await invoiceService.getInvoiceForBatch(batchId, invoiceId);
 
```

Comment:
In some other recent work (I think creating/modifying user permissions) we had access/permissions checks at the water service layer.
Should we do similar for all billing routes?

Reply:
We don't for the creation of a batch at the moment. But we should do this to this and other routes as part of a story to make better use of the JWT token I think.

Reply:
Sounds good